src/geocat/comp/climatology.py

```diff
@@ -322,3 +326,144 @@ def month_to_season(
     # Therefore, we must filter out the NaNs
     return means.sel(
         {time_coord_name: means[time_coord_name].dt.month == months[1]})
+
+
+def time_avg(dset, window, time_dim=None, rolling=False, **rolling_kwargs):
+    """Function to take N-d data and compute averages over the time dimension
+    using a specified window size.
+
+    Parameters
+    ----------
+    dset : `xarray.Dataset`, `xarray.DataArray`, `numpy.ndarray`
+        The data on which to operate
+    window : `int`
+        Size of the window used to compute the averages
+    time_dim : `str`, `int`
+        Name of the time coordinate for `xarray` objects. If `None`, then the coordinate with
+        Datetime objects will be used. If `dset` is a numpy array, `time_dim` must be the
+        index of the dimension that is represented by time.
+    rolling : `boolean`, Optional
+        Defaults to `False` and computes a sequential average where the
+        stride of the window is equal to the window size. `True` computes a
+        rolling average where the window uses a stride of 1.
+    **rolling_kwargs :
+        Keyword arguments to pass into `xarray.DataArray.rolling`.
+
+    Returns
+    -------
+    computed_dset: same type as dset
+        The computed data
+    """
+
+    min_periods = rolling_kwargs.get('min_periods')
+    center = rolling_kwargs.get('center')
+
+    if isinstance(dset, np.ndarray):  # TODO: duck typing for arrays
+        if time_dim == None or not isinstance(time_dim, int):
```

Comment:
Use `time_dim` is None rather than `time_dim == None`
```suggestion
        if time_dim is None or not isinstance(time_dim, int):
```

Reply:
@andersy005 this is a draft PR from my intern, she and I will do a code review before we take this out of draft, and then we will request reviews once it is a non-draft PR. 

Until then, she will be actively developing, and there's no need to comment on the occasional weirdness that happens in early draft phases.

Reply:
FYI I added this suggestion as part of one of my other commits. This can be labeled as resolved now.